server/app/mutations/grid_services/common.rb

```diff
@@ -84,6 +86,30 @@ def build_grid_service_secrets(existing_secrets)
       service_secrets
     end
 
+    def build_service_volumes()
+      service_volumes = []
+      self.volumes.each do |vol|
+        vol_spec = parse_volume(vol)
+        if vol_spec[:volume]
+          # Named volume, try to find the proper mapping for external volumes
+          # NULL stack services don't have revs, for those just use the "global" name
+          volume_name = vol_spec[:volume]
+          if self.stack.latest_rev
```

Comment:
The service mutations do not currently get `latest_rev` as input. (Or am I blind? :D ) Also the service mutations are actually called from multiple places (`Stack::Create`, `Stack::Update`, `Stack::Deploy`, ...) so there would be multiple places to "fix" to give that as input. In that light I see the "backlinking" actually quite clean solution.